src/EFCore.Cosmos/Query/Internal/CosmosQueryMetadataExtractingExpressionVisitor.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq.Expressions;
+
+namespace Microsoft.EntityFrameworkCore.Cosmos.Query.Internal
+{
+    public class CosmosQueryMetadataExtractingExpressionVisitor : ExpressionVisitor
+    {
+        private readonly CosmosQueryCompilationContext _cosmosQueryCompilationContext;
+
+        public CosmosQueryMetadataExtractingExpressionVisitor(CosmosQueryCompilationContext cosmosQueryCompilationContext)
+        {
+            _cosmosQueryCompilationContext = cosmosQueryCompilationContext;
+        }
+
+        protected override Expression VisitMethodCall(MethodCallExpression methodCallExpression)
+        {
+            if (methodCallExpression.Method.DeclaringType == typeof(CosmosQueryableExtensions)
+                && methodCallExpression.Method.Name == nameof(CosmosQueryableExtensions.WithPartitionKey))
```

Comment:
You can have a stronger condition here
```suggestion
            if (methodCallExpression.Method == CosmosQueryableExtensions.WithPartitionKeyMethodInfo)
```

Reply:
I tested: `if (methodCallExpression.Method == CosmosQueryableExtensions.WithPartitionKeyMethodInfo)` 

... and it resolves to false when hitting the WithPartitionKey method. The original resolves to `true` under same circumstances.

I'd be happy yo take a closer look later. It's 1:00am in my time-zone at the moment.

Reply:
Updated suggestion

Reply:
👍